### PR TITLE
Knative Backstage plugin article

### PR DESCRIPTION
**Pull request by aliok**
Preview https://deploy-preview-5897--knative.netlify.app/blog/articles/knative-backstage-plugins/

### CODE REVIEW

File: blog/docs/articles/knative-backstage-plugins.md
Comment (creydr) [inline blog/docs/articles/knative-backstage-plugins.md:16]:
```suggestion
The plugins are the frontend part of the Backstage plugin. They are responsible for rendering the UI and communicating with the backend. The backend is responsible for talking to the Kubernetes API server and providing the necessary information to the frontend. Currently, we only have one plugin and it requires a backend to work. We think that most of the plugins we might provide in the future will require a backend-for-frontend to work.
```

nit

File: blog/docs/articles/knative-backstage-plugins.md
Comment (creydr) [inline blog/docs/articles/knative-backstage-plugins.md:26]:
```suggestion
The plugin fetches Knative EventType, Broker, and Trigger resources from the backend. Actually, it doesn't fetch the resources as is since the Kubernetes resources are not particularly designed for using them in this kind of UI, but it fetches better suited representations of the resources where some fields are ignored, some aggregated, and some transformed.
```
nit

File: blog/docs/articles/knative-backstage-plugins.md
Comment (creydr) [inline blog/docs/articles/knative-backstage-plugins.md:31]:
```suggestion
Knative Triggers can be pointing at any [Addressable](https://knative.dev/docs/concepts/duck-typing/#addressable) or at even an URL directly. Since Backstages side won't know how to handle these, the plugin will only show the trigger's subscriber if it is already registered in Backstage. For this relation, we use the [`backstage.io/kubernetes-id`](https://backstage.io/docs/features/kubernetes/configuration#surfacing-your-kubernetes-components-as-part-of-an-entity) annotation in the Backstage entity and the Kubernetes resource.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (creydr) [inline blog/docs/articles/knative-backstage-plugins.md:42]: Do we have a link for some how-tos for this?

**Review by creydr — COMMENTED** [review]
Nice. :+1: 
Thanks @aliok.

I only left some nits. But looks great so far

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:7]: Adding `{:target="_blank"}` at the end of the link will make this page open in a new tab, so that blog post tab won't be lost!

Just a nit, to have a better reader experience!
```suggestion
[Backstage](https://backstage.io/){:target="_blank"} is a platform for building developer portals. It provides a unified way to manage and visualize the different resources that developers use in their daily work. 
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:20]:
```suggestion
[The Event Mesh plugin](https://github.com/knative-extensions/backstage-plugins){:target="_blank"} is a Backstage plugin that allows you to view and manage Knative Eventing resources. The plugin talks to a special backend that runs in the Kubernetes cluster and communicates with the Kubernetes API server. 
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:24]:
```suggestion
This plugin leverages Backstage's [entity provider](https://backstage.io/docs/features/software-catalog/external-integrations/#custom-entity-providers){:target="_blank"} and [entity processor](https://backstage.io/docs/features/software-catalog/external-integrations/#custom-processors){:target="_blank"} concept. The entity provider is responsible for fetching the resources from the backend and the entity processor is responsible for processing the resources and making them available to the frontend. The provider is the part that talks to the custom backend.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:31]:
```suggestion
Knative Triggers can be pointing at any [Addressable](https://knative.dev/docs/concepts/duck-typing/#addressable){:target="_blank"} or at even a URL directly. Since Backstage side won't know how to handle these, the plugin will only show the trigger's subscriber if it is already registered in Backstage. For this relation, we use the [`backstage.io/kubernetes-id`](https://backstage.io/docs/features/kubernetes/configuration#surfacing-your-kubernetes-components-as-part-of-an-entity){:target="_blank"} annotation in the Backstage entity and the Kubernetes resource.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:33]:
```suggestion
How the matching works is documented with diagrams in the [Event Mesh plugin README file](https://github.com/knative-extensions/backstage-plugins?tab=readme-ov-file#event-mesh-plugin-1){:target="_blank"}.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:44]:
```suggestion
However, if you want a quick look at the plugin, you can check out the [demo video](https://www.youtube.com/watch?v=4h1j1v8KrY0){:target="_blank"}.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:45]:
```suggestion
The demo video is recorded with the quick start available in Ali Ok's [demo repository](https://github.com/aliok/knative-backstage-demo){:target="_blank"}.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:52]:
```suggestion
We are looking for contributors to help us improve the plugin and the backend. If you are interested in contributing, please check out the [README file](https://github.com/knative-extensions/backstage-plugins){:target="_blank"} of the plugins repository. How to start the backend, how to install the plugin, and how to modify the plugin are all documented there.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:54]:
```suggestion
There are a few issues that are marked as good first issues and we are looking for help with them. If you are interested in contributing, please check out the [good first issues](https://github.com/knative-extensions/backstage-plugins/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22){:target="_blank"}.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:58]:
```suggestion
We are exploring the possibility of adding more functionality to integrate Backstage with Knative. One aspect here is reusing the [Knative Functions](https://knative.dev/docs/functions/){:target="_blank"} templates as [Backstage templates](https://backstage.io/docs/features/software-templates/adding-templates){:target="_blank"}. This would allow developers to create new Knative Functions from Backstage and manage them from there, while having the compatibility with the [Func CLI](https://knative.dev/docs/functions/install-func/){:target="_blank"}.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [inline blog/docs/articles/knative-backstage-plugins.md:62]:
```suggestion
If you have any questions or feedback, please feel free to reach out to us. You can find us in the [CNCF Slack](https://cncf.slack.io){:target="_blank"} in the [#knative](https://cloud-native.slack.com/archives/C04LGHDR9K7){:target="_blank"} channel.
```

**Review by Leo6Leo — COMMENTED** [review]
Just some nit, otherwise it looks good and easy to follow along!

File: blog/docs/articles/knative-backstage-plugins.md
Comment (Leo6Leo) [reply blog/docs/articles/knative-backstage-plugins.md:42]: I think here is the link to the tutorial on how to install the demo https://github.com/aliok/knative-backstage-demo?tab=readme-ov-file#installation

And how to run the backend plugin
https://github.com/knative-extensions/backstage-plugins?tab=readme-ov-file#running-the-backstage-plugin

Comment (aliok) [reply blog/docs/articles/knative-backstage-plugins.md]: Added a link now

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [reply blog/docs/articles/knative-backstage-plugins.md]: Thanks, updated

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [reply blog/docs/articles/knative-backstage-plugins.md]: Thanks, appliec manually

**Comment by aliok** [comment]
@Leo6Leo @creydr can you have a look again?

File: blog/docs/articles/knative-backstage-plugins.md
Comment (nainaz) [inline blog/docs/articles/knative-backstage-plugins.md:9]: I think we should drop the use of However as it is there twice in the first paragraph here and may be try something like "While Backstage is not designed as a next generation Kubernetes Dashboard , it can visualize and partially manage Knative resources.  These resources would be read-only and focused on what's relevant to developers.

File: blog/docs/articles/knative-backstage-plugins.md
Comment (nainaz) [inline blog/docs/articles/knative-backstage-plugins.md:11]: I think we should drop the use of However as it is there twice in the first paragraph here and may be try something like "`While Backstage is not designed as a next generation Kubernetes Dashboard , it can visualize and partially manage Knative resources. These resources would be read-only and focused on what's relevant to developers.`

File: blog/docs/articles/knative-backstage-plugins.md
Comment (nainaz) [inline blog/docs/articles/knative-backstage-plugins.md:22]: This whole part seems to be the repeat of earlier intro paragraph. Should we eliminate it?

File: blog/docs/articles/knative-backstage-plugins.md
Comment (nainaz) [inline blog/docs/articles/knative-backstage-plugins.md:26]: This reads contradictory as first sentence said fetches and second says it doesn't fetch. :( 
` The plugin fetches Knative EventType, Broker, and Trigger resources from the backend. Instead of directly fetching the raw Kubernetes resources, it gathers a more user-friendly representation of the data. This means certain fields are excluded, some are combined for clarity, and others are transformed to be easier to understand within this interface.`

File: blog/docs/articles/knative-backstage-plugins.md
Comment (nainaz) [inline blog/docs/articles/knative-backstage-plugins.md:44]: For a quicker look, check out the demo video

**Review by nainaz — COMMENTED** [review]
Such a great write up. Left some nit-pick.

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [inline blog/docs/articles/knative-backstage-plugins.md:9]:
```suggestion
While Backstage is not designed as a next generation Kubernetes Dashboard, it can visualize and partially manage Knative resources. These resources would be read-only and focused on what's relevant to developers.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [inline blog/docs/articles/knative-backstage-plugins.md:11]:
```suggestion
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [inline blog/docs/articles/knative-backstage-plugins.md:21]:
```suggestion
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [inline blog/docs/articles/knative-backstage-plugins.md:24]:
```suggestion
The plugin fetches Knative EventType, Broker, and Trigger resources from the backend. Instead of directly fetching the raw Kubernetes resources, it gathers a more user-friendly representation of the data. This means certain fields are excluded, some are combined for clarity, and others are transformed to be easier to understand within this interface.
```

File: blog/docs/articles/knative-backstage-plugins.md
Comment (aliok) [inline blog/docs/articles/knative-backstage-plugins.md:42]:
```suggestion
However, for a quicker look at the plugin, you can check out the [demo video](https://www.youtube.com/watch?v=4h1j1v8KrY0){:target="_blank"}.
```

**Comment by aliok** [comment]
@Leo6Leo @creydr @nainaz  can you have a look again and lgtm if good?

**Review by creydr — APPROVED** [review]
Thanks a lot for this @aliok 

/lgtm

**Comment by nainaz** [comment]
Looks like I am too late :) 
Thank you for writing this @aliok

**Pull request merged**
